scripts/core/platform.yml

```diff
@@ -242,3 +242,40 @@ etors:
     - name: HIP
       value: "4"
       desc: "The backend is HIP"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Extension name maximum length."
+name: $X_MAX_EXTENSION_NAME_LENGTH
+value: "256"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Extension properties."
+name: $x_extension_properties_t
+base: $x_base_properties_t
+class: $xPlatform
+members:
+    - type: char
+      name: name[$X_MAX_EXTENSION_NAME_LENGTH]
+      desc: "[in] null-terminated extension name."
+    - type: uint32_t
+      name: version
+      desc: "[in] version of the extension using $X_MAKE_VERSION."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Retrieve the set of supported extensions."
+class: $xPlatform
+name: GetExtensionProperties
```

Comment:
Currently we have a `PlatformGetInfo` query for supported extensions but it returns a list of names in a string with some delimiter. I'm not convinced we need an additional entry point, we could return an array of structs from `PlatformGetInfo` instead of a list of names in a string.

Reply:
sounds like a good plan 👍